src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -0,0 +1,89 @@
+package org.rust.ide.hints
+
+import com.intellij.codeInsight.hints.HintInfo
+import com.intellij.codeInsight.hints.InlayInfo
+import com.intellij.codeInsight.hints.InlayParameterHintsProvider
+import com.intellij.codeInsight.hints.Option
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.valueParameters
+import org.rust.lang.core.types.infer.inferDeclarationType
+import org.rust.lang.core.types.ty.TyUnit
+import org.rust.lang.core.types.ty.TyUnknown
+
+private val RsCallExpr.declaration: RsFunction?
+    get() = (expr as? RsPathExpr)?.path?.reference?.resolve() as? RsFunction
+
+private val RsMethodCallExpr.declaration: RsFunction?
+    get() = reference.resolve() as? RsFunction
+
+enum class HintType(desc: String, enabled: Boolean) {
+    LET_BINDING_HIST("Show local variable type hints", true) {
+        override fun provideHints(elem: PsiElement): List<InlayInfo> {
+            val element = elem as? RsLetDecl ?: return emptyList()
+            if (element.typeReference != null) return emptyList()
+            val ident = element.pat as? RsPatIdent ?: return emptyList()
+            val patBinding = ident.patBinding
+            val type = inferDeclarationType(patBinding)
+            if (type is TyUnknown || type is TyUnit) return emptyList()
+            return listOf(InlayInfo(": " + type.toString(), patBinding.textRange.endOffset))
+        }
+
+        override fun isApplicable(elem: PsiElement): Boolean
+            = elem is RsLetDecl
+    },
+    PARAMETER_HINT("Show argument name hints", true) {
+        override fun provideHints(elem: PsiElement): List<InlayInfo> {
+            val (declaration, valueArgumentList) = when (elem) {
+                is RsCallExpr -> (elem.declaration to elem.valueArgumentList)
+                is RsMethodCallExpr -> (elem.declaration to elem.valueArgumentList)
+                else -> null
+            } ?: return emptyList()
+            return declaration?.valueParameters
+                ?.mapIndexedNotNull { i, parameter -> transform(i, parameter, valueArgumentList) }
+                ?: emptyList()
+        }
+
+        override fun isApplicable(elem: PsiElement): Boolean
+            = elem is RsCallExpr || elem is RsMethodCallExpr
+
+        fun transform(index: Int, valueParameter: RsValueParameter, valueArgumentList: RsValueArgumentList): InlayInfo? {
+            if (index >= valueArgumentList.exprList.size)
+                return null
+            return InlayInfo(valueParameter.pat?.text + ":", valueArgumentList.exprList[index].textRange.startOffset)
```

Comment:
You also need to return here if `valueParameter.pat?` is `null` here to handle anonymous arguments properly. Here's a snippet to test this case:

```rust
trait Foo {
    fn foo(&self, u32);
    fn bar(&self) {
        self.foo(10);    // <-- shouldn't add `null: `
    }
}
```